vim-patch:8.1.2297: the ex_vimgrep() function is too long

zeertzjq opened the pull request:
#### vim-patch:8.1.2297: the ex_vimgrep() function is too long

Problem:    The ex_vimgrep() function is too long.
Solution:   Split it in three parts. (Yegappan Lakshmanan, closes vim/vim#5211)
https://github.com/vim/vim/commit/d6a98a3a9768568b668f91a53267b36f86b84466

Including a missing change to ex_vimgrep() from patch 8.0.1831.

seandewar commented on src/nvim/quickfix.c:
```diff
@@ -5328,118 +5337,85 @@ static bool existing_swapfile(const buf_T *buf)
+    args->tomatch = MAXLNUM;
   }
 
-  // Get the search pattern: either white-separated or enclosed in //
```
Keep this comment

zeertzjq replied:
Oops. Thanks!

seandewar commented on src/nvim/quickfix.c:
```diff
@@ -5522,36 +5504,87 @@ void ex_vimgrep(exarg_T *eap)
+
+  win_T *wp = NULL;
+  qf_info_T *qi = qf_cmd_get_or_alloc_stack(eap, &wp);
+  char_u *target_dir = NULL;
```
This can be moved further down to where `status` is

zeertzjq replied:
It needs to be initialized before the first `goto`.

seandewar replied:
Ah, sorry; missed that.

seandewar commented on src/nvim/quickfix.c:
```diff
@@ -5522,36 +5504,87 @@ void ex_vimgrep(exarg_T *eap)
+  int status = vgr_process_files(wp, qi, &args, &redraw_for_dummy, &first_match_buf, &target_dir);
+
+  if (status != OK) {
+    FreeWild(args.fcount, args.fnames);
+    decr_quickfix_busy();
+    goto theend;
+  }
+
+  FreeWild(args.fcount, args.fnames);
```
Doesn't really matter, but Vim could've just called FreeWild before the `if` 😄

zeertzjq replied:
I think I'll keep this.

zeertzjq replied:
I think I'll keep this.

seandewar replied:
No problem, was just an observation :+1:

zeertzjq committed:
vim-patch:8.1.2297: the ex_vimgrep() function is too long

Problem:    The ex_vimgrep() function is too long.
Solution:   Split it in three parts. (Yegappan Lakshmanan, closes vim/vim#5211)
https://github.com/vim/vim/commit/d6a98a3a9768568b668f91a53267b36f86b84466

Including a missing change to ex_vimgrep() from patch 8.0.1831.

seandewar reviewed (approved):
LGTM!
